src/espcomp/mod.rs

```diff
@@ -0,0 +1,129 @@
+use std::path::PathBuf;
+
+use anyhow::{Context, Result};
+use flate2::read::GzDecoder;
+use log::info;
+use tar::Archive;
+
+pub mod comp;
+
+const API_BASE_URL: &str = "https://api.components.espressif.com";
+
+struct ApiClient {
+    base_url: String,
+}
+
+impl ApiClient {
+    pub fn new() -> Self {
+        Self {
+            base_url: API_BASE_URL.to_string(),
```

Comment:
Nit: what is the point of having this variable, if you don't provide a constructor, which takes a different `base_url`? If you plan to provide one, disregard. Otherwise I don't see why we need the `base_url` var in the first place.

Reply:
I can remove this for now, but think it's a good idea to keep the functions belonging to the `ApiClient` impl in case someone later needs to configure the client to use a local proxy / mirror or something like that (rather than having them just be bare functions).